RockLib.Messaging.Http/HttpClientSender.cs

```diff
@@ -13,36 +13,75 @@ namespace RockLib.Messaging.Http
     /// An implementation of <see cref="ISender" /> that sends messages with an
     /// <see cref="HttpClient"/>.
     /// </summary>
+#pragma warning disable CS1069
     public class HttpClientSender : ISender
+#pragma warning restore CS1069
     {
-        private readonly HttpContentHeaders _defaultContentHeaders = new ByteArrayContent(new byte[0]).Headers;
+        private readonly HttpContentHeaders _defaultContentHeaders = new ByteArrayContent(Array.Empty<byte>()).Headers;
         private readonly HttpRequestHeaders _defaultRequestHeaders = new HttpRequestMessage().Headers;
 
         private readonly HttpClient _client;
 
+        private bool disposedValue;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpClientSender"/> class.
+        /// </summary>
+        /// <param name="name">The name of the sender.</param>
+        /// <param name="url">The url to send messages to.</param>
+        /// <param name="method">The http method to use when sending messages.</param>
+        /// <param name="defaultHeaders">Default headers that are added to each http request.</param>
+        public HttpClientSender(string name, Uri url, string method = "POST", IReadOnlyDictionary<string, string>? defaultHeaders = null)
+        {
+            Name = name ?? throw new ArgumentNullException(nameof(name));
+            Url = url is not null ? url.OriginalString : throw new ArgumentNullException(nameof(url));
+            Method = new HttpMethod(method ?? throw new ArgumentNullException(nameof(method)));
+
+            _client = new HttpClient();
+
+            if (defaultHeaders is not null)
```

Comment:
You could maybe refactor part of the constructor code to reduce duplication since parts are used in other constructors.